### ISSUE TITLE
zstd-jni 1.4.9-2/1.4.9-3 are not compatible with Java 8 (regression)

### ISSUE THREAD

**Issue by iemejia** [luben/zstd-jni#171]
While trying to upgrade to the 1.4.9-2 version we noticed that it was looking a method that it does not exist in Java 8

```
2021-04-04T08:54:02.3760587Z java.lang.NoSuchMethodError: java.nio.ByteBuffer.clear()Ljava/nio/ByteBuffer;
2021-04-04T08:54:02.3765986Z 	at com.github.luben.zstd.RecyclingBufferPool.release(RecyclingBufferPool.java:50)
2021-04-04T08:54:02.3772162Z 	at com.github.luben.zstd.ZstdOutputStreamNoFinalizer.close(ZstdOutputStreamNoFinalizer.java:276)
2021-04-04T08:54:02.3775534Z 	at org.apache.avro.file.ZstandardCodec.compress(ZstandardCodec.java:75)
2021-04-04T08:54:02.3781121Z 	at org.apache.avro.file.DataFileStream$DataBlock.compressUsing(DataFileStream.java:386)
2021-04-04T08:54:02.3784998Z 	at org.apache.avro.file.DataFileWriter.writeBlock(DataFileWriter.java:407)
2021-04-04T08:54:02.3789879Z 	at org.apache.avro.file.DataFileWriter.writeIfBlockFull(DataFileWriter.java:351)
2021-04-04T08:54:02.3793796Z 	at org.apache.avro.file.DataFileWriter.append(DataFileWriter.java:320)
2021-04-04T08:54:02.3798063Z 	at org.apache.avro.TestDataFile.testGenericWrite(TestDataFile.java:115)
2021-04-04T08:54:02.3802146Z 	at org.apache.avro.TestDataFile.runTestsInOrder(TestDataFile.java:96)
2021-04-04T08:54:02.3805805Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-04-04T08:54:02.3810720Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2021-04-04T08:54:02.3819179Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-04-04T08:54:02.3890907Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2021-04-04T08:54:02.3893392Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
```

This is a classic error that happens when compiling the code with a Java 11 compiler, so maybe worth a new release to fix compatibility with Java 8 clients.

**Comment by luben** [comment]
Hmm, I see. I will look how to fix it. In Java 8 and earlier `clear` is method on `Buffer` but later versions moved it to be method of `ByteBuffer`. So may be there is something with the generated bytecode that does not look for inherited methods.

**Comment by luben** [comment]
Let's see if it fixes it. I will push new release today or tomorrow.

**Comment by luben** [comment]
@iemejia , I just pushed v1.4.9-3 to maven central. Let me know if the problem is solved.

**Comment by iemejia** [comment]
Thanks for the quick release, it seems however that it is still broken:
```
2021-04-05T09:29:11.5206875Z java.lang.NoSuchMethodError: java.nio.ByteBuffer.limit(I)Ljava/nio/ByteBuffer;
2021-04-05T09:29:11.5212510Z 	at com.github.luben.zstd.RecyclingBufferPool.release(RecyclingBufferPool.java:53)
2021-04-05T09:29:11.5218100Z 	at com.github.luben.zstd.ZstdOutputStreamNoFinalizer.close(ZstdOutputStreamNoFinalizer.java:276)
2021-04-05T09:29:11.5223028Z 	at org.apache.avro.file.ZstandardCodec.compress(ZstandardCodec.java:75)
2021-04-05T09:29:11.5227720Z 	at org.apache.avro.file.DataFileStream$DataBlock.compressUsing(DataFileStream.java:386)
2021-04-05T09:29:11.5232851Z 	at org.apache.avro.file.DataFileWriter.writeBlock(DataFileWriter.java:407)
2021-04-05T09:29:11.5238500Z 	at org.apache.avro.file.DataFileWriter.writeIfBlockFull(DataFileWriter.java:351)
2021-04-05T09:29:11.5269962Z 	at org.apache.avro.file.DataFileWriter.append(DataFileWriter.java:320)
```

**Comment by iemejia** [comment]
If you are compiling with a new JDK > 9 you should explicitly pass the `--release` option to target the proper bytecode.
https://stackoverflow.com/questions/43102787/what-is-the-release-flag-in-the-java-9-compiler

**Comment by luben** [comment]
I see, following with a new release shortly.

**Comment by luben** [comment]
@iemejia ,  I released v1.4.9-4 that is compiled with the `--release` flag. On top of `-source` and `-target` it adds bootstrap path to the library version used. Thanks for the hint!

Can you try it?

**Comment by iemejia** [comment]
Everything working now on 1.4.9-4! Thanks @luben
